### DISCUSSION TITLE
Integrating Static Quick Actions with Swift Composable Architecture

### DISCUSSION THREAD

**Discussion by ammerzon** [pointfreeco/swift-composable-architecture#3261] [Q&A]
I'm working on implementing static quick actions in an iOS app built with Swift Composable Architecture (SCA). While I've successfully added the necessary entries to the `Info.plist` file, I'm encountering challenges in handling these actions within the SCA paradigm.

<img src="https://github.com/user-attachments/assets/88aa8de5-5a53-4120-9517-7d957ec85ab0" width="500">

Typically, quick actions are managed through a `SceneDelegate`, as demonstrated in [various tutorials](https://medium.com/@amnmustafa15/using-static-quick-actions-to-navigate-to-different-swiftui-views-dcbcc9661744):

```swift
class SceneDelegate: NSObject, UIWindowSceneDelegate {
    private let qaService = QAService.shared
    
    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
        if let shortcutItem = connectionOptions.shortcutItem {
            qaService.action = QA(shortcutItem: shortcutItem)
        }
    }
    
    func windowScene(_ windowScene: UIWindowScene, performActionFor shortcutItem: UIApplicationShortcutItem, completionHandler: @escaping (Bool) -> Void) {
        qaService.action = QA(shortcutItem: shortcutItem)
        completionHandler(true)
    }
}
```

However, I'm curious about the best practices for integrating this functionality within the SCA framework. Has anyone successfully implemented static quick actions with SCA? I'm particularly interested in:

- How to handle the lifecycle events typically managed by SceneDelegate in a SCA-compliant manner.
- Strategies for routing quick actions to the appropriate reducers and effects.

**Comment by maksimn** [comment]
Hello, @ammerzon 
I suppose in the `SceneDelegate` you should have a reference to SCA `Store` or some entity that can dispatch SCA actions.
And you should define some SCA actions corresponding to quick actions. 
After that you can send this SCA actions in the `SceneDelegate` and handle them in reducers/effects.

**Reply by ammerzon** [reply]
Thank you, @maksimn, for your response. I've considered that approach, but I'm unsure how to integrate the SCA `store` into the `SceneDelegate` or an alternative component to facilitate action dispatching.
